internal/types/account_test.go

```diff
@@ -43,24 +64,120 @@ func TestNewAccountFromSignerWithAddress(t *testing.T) {
+
 	assert.Equal(t, AccountZero, account.Address)
+	assert.Equal(t, AccountZero, account.AccountAddress())
+	assert.Equal(t, key.AuthKey(), account.AuthKey())
+	assert.Equal(t, key.PubKey(), account.PubKey())
 }
 
 func TestNewAccountFromSignerWithAddressMulti(t *testing.T) {
 	key, err := crypto.GenerateEd25519PrivateKey()
 	assert.NoError(t, err)
 
-	authenticationKey := crypto.AuthenticationKey{}
+	_, err = NewAccountFromSigner(key, AccountZero, AccountOne)
+	assert.Error(t, err)
+}
+
+type WrapperSigner struct {
+	signer crypto.Signer
+}
+
+func (w *WrapperSigner) Sign(_ []byte) (*crypto.AccountAuthenticator, error) {
+	return nil, errors.New("not implemented")
+}
+func (w *WrapperSigner) SignMessage(_ []byte) (crypto.Signature, error) {
+	return nil, errors.New("not implemented")
+}
+func (w *WrapperSigner) SimulationAuthenticator() *crypto.AccountAuthenticator {
+	return nil
+}
+func (w *WrapperSigner) AuthKey() *crypto.AuthenticationKey {
+	return &crypto.AuthenticationKey{}
+}
+func (w *WrapperSigner) PubKey() crypto.PublicKey {
+	// Note this is just for testing
+	return &crypto.Ed25519PublicKey{}
+}
+
+func TestAccount_ExtractMessageSigner(t *testing.T) {
+	ed25519PrivateKey, err := crypto.GenerateEd25519PrivateKey()
+	assert.NoError(t, err)
+	ed25519Account, err := NewAccountFromSigner(ed25519PrivateKey)
+	assert.NoError(t, err)
+
+	ed25519Out, ok := ed25519Account.ExtractMessageSigner()
+	assert.True(t, ok)
+	assert.Equal(t, ed25519PrivateKey, ed25519Out)
+
+	ed25519SingleSignerAccount, err := NewAccountFromSigner(crypto.NewSingleSigner(ed25519PrivateKey))
+	assert.NoError(t, err)
+
+	ed25519Out, ok = ed25519SingleSignerAccount.ExtractMessageSigner()
+	assert.True(t, ok)
+	assert.Equal(t, ed25519PrivateKey, ed25519Out)
+
+	secp256k1PrivateKey, err := crypto.GenerateSecp256k1Key()
+	assert.NoError(t, err)
+	secp256k1SingleSignerAccount, err := NewAccountFromSigner(crypto.NewSingleSigner(secp256k1PrivateKey))
+	assert.NoError(t, err)
+
+	secp256k1Out, ok := secp256k1SingleSignerAccount.ExtractMessageSigner()
+	assert.True(t, ok)
+	assert.Equal(t, secp256k1PrivateKey, secp256k1Out)
+
+	wrapperSigner := &WrapperSigner{signer: secp256k1SingleSignerAccount}
+	customAccount, err := NewAccountFromSigner(wrapperSigner)
+	assert.NoError(t, err)
+	out, ok := customAccount.ExtractMessageSigner()
+	assert.False(t, ok)
+	assert.Nil(t, out)
+}
+
+func TestAccount_ExtractPrivateKeyString(t *testing.T) {
+	ed25519PrivateKey, err := crypto.GenerateEd25519PrivateKey()
+	assert.NoError(t, err)
+	ed25519Account, err := NewAccountFromSigner(ed25519PrivateKey)
+	assert.NoError(t, err)
 
-	_, err = NewAccountFromSigner(key, authenticationKey, authenticationKey)
+	ed25519KeyString, err := ed25519Account.ExtractPrivateKeyString()
+	assert.NoError(t, err)
+	expectedEd25519String, err := ed25519PrivateKey.ToAIP80()
+	assert.NoError(t, err)
+	assert.Equal(t, expectedEd25519String, ed25519KeyString)
```

Comment:
Maybe add another test case to verify [AIP 80](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-80.md)

`assert.True(t, strings.HasPrefix(ed25519KeyString, "ed25519-priv-"))`

Reply:
I'm pretty sure this is testing it right here that it's AIP-80